### ISSUE TITLE
Intermittent ORA-28547 error occurring 

### ISSUE THREAD

**Issue by mjdavies** [rana/ora#217]
Hi all

Over the last couple of days we've seen a stable service start throwing intermittent ORA-28547 errors.

Just wondering if anyone else has seen this type of thing before, or how I can debug what's going on?

The service goes through about 3 requests per second to the oracle server normally, and then one or two will fail, then carry on, it's very odd.

Any help, greatly appreciated.

If you need any more information from me please ask.

**Comment by tgulacsi** [comment]
http://www.dba-oracle.com/t_ora_28547_connection_to_server_failed_probable_oracle_net_admin_error.htm says it's an intermittent network error.
I'm adding this ORA-28547 to maybeBadConn, so it'll be retried by database/sql.

**Comment by mjdavies** [comment]
Hi there.

We found the issue in the end.  We weren't closing the rows in the query, so the number of sessions showing on the oracle database side just carried on going up and up.  As the oracle database server got busier and busier, in the end it just didn't have the resource to keep all the sessions open so started dropping them, which was throwing errors our side.

It seems to be a problem that we've isolated to oracle db, in that it doesn't happen with mssql or mysql, maybe oracle handles open idle sessions differently to mssql or mysql, or maybe those settings can be configured on the oracle server.

We're not the admins of the oracle server so it's very hard to tell.

Anyway, since we started closing the rows, the session numbers don't keep going up, and so far, not one error since.

Thanks for getting back to us anyway, but just to clarify, it's not an intermittent network error.
